### PR TITLE
Split compiler and core

### PR DESCRIPTION
**Pull request by tbosch**
Note: The first 4 commits are from https://github.com/angular/angular/pull/18720

### CODE REVIEW

File: packages/compiler/src/aot/static_reflector.ts [hunk 1]
```diff
@@ -266,65 +262,40 @@ export class StaticReflector implements CompileReflector {
...
-    this._registerDecoratorOrConstructor(
-        this.findDeclaration(ANGULAR_CORE, 'HostBinding'), HostBinding);
-    this._registerDecoratorOrConstructor(
-        this.findDeclaration(ANGULAR_CORE, 'HostListener'), HostListener);
-    this._registerDecoratorOrConstructor(
-        this.findDeclaration(ANGULAR_CORE, 'Directive'), Directive);
-    this._registerDecoratorOrConstructor(
-        this.findDeclaration(ANGULAR_CORE, 'Component'), Component);
-    this._registerDecoratorOrConstructor(this.findDeclaration(ANGULAR_CORE, 'NgModule'), NgModule);
+    this._registerDecoratorOrConstructor(MetadataIdentifier.Host, createHost);
```
Comment (chuckjaz) [inline packages/compiler/src/aot/static_reflector.ts]: As discussed offline, I don't believe `MetadataIdentifer` is required.

File: packages/platform-browser-dynamic/src/compiler_reflector.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
+/**
```
Comment (chuckjaz) [inline packages/platform-browser-dynamic/src/compiler_reflector.ts]: Consider moving the `reflect_metadata` requirement to this module so we no longer require it in `compiler`.

**Review by chuckjaz — CHANGES_REQUESTED** [review]

**Review by chuckjaz — APPROVED** [review]

File: packages/platform-browser-dynamic/src/compiler_reflector.ts (hunk 2 above)
Comment (tbosch) [inline packages/platform-browser-dynamic/src/compiler_reflector.ts]: I would like to postpone this to another PR...

**Comment by tbosch** [comment]
Note: This was already green, CI only runs because of another rebase.

**Review by mhevery — APPROVED** [review]

**Pull request closed without merging**